### ISSUE TITLE
test functorch under a debug build of Python

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#509]
Suggested by @albanD (we want to make sure we're not leaking memory or doing something weird with the pybind API) who volunteered to help with this.

Here are the instructions on how to build functorch and run tests:

```
# Step 1: Build latest PyTorch, somehow. (NB: not in debug mode, some of our tests fail under that)
...

# Step 2: Build functorch
git clone https://github.com/pytorch/functorch.git
cd functorch
python setup.py develop

# Step 3: run tests

# Testing dependencies. Hopefully these are python-only?
pip install expecttest

# basically, all the tests aside from test_functorch_lagging_op_db are supposed to pass
# assuming our CI is green
# you might get some failures from "networkx is not available" but installing networkx requires having a build of numpy...
pytest -v --ignore=test/test_functorch_lagging_op_db.py test/ -v
```

**Comment by albanD** [comment]
Just for the record, building pytorch at 79a216ce576521aebb913d7c5e72e1f36fb04a86 with python 3.8 with pydebug at 66dd5338a1ca98921c8e6c51228541ef8ed8076a

Summary is 15 failed, 5431 passed, 5018 skipped, 376 xfailed, 493 warnings.
One of the error is due to networkx.
The 14 others are `RuntimeError: DispatchKeys not yet supported: FuncTorchDynamicLayerBackMode FuncTorchDynamicLayerFrontMode`

So all seems good here!

**Comment by zou3519** [comment]
Thanks @albanD !
